device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -420,6 +420,11 @@ public void sendReportedProperties(Set<Property> reportedProperties, Integer ver
      * with a status and a reason why the device's status changed. When the callback is fired, the provided context will
      * be provided alongside the status and reason.
      *
+     * This connection status callback is not triggered by any upstream connection change events. For example, if
```

Comment:
Are you also planning customer-facing docs?

Reply:
What do you mean? These are the customer facing docs. This class is "InternalClient" but the method and its javadocs are exposed via inheritance in both ModuleClient and DeviceClient

Reply:
I assumed the fact it is named internal client meant it wasn't customer facing. My mistake. :)